### PR TITLE
quality_data: overview: Fix incorrect field summing passing grants

### LINKED ISSUE #246: Quality Dashboard shows > 100% org IDs
[By grants view](https://qualitydashboard.threesixtygiving.org/alldata#grants) showing 108% Includes at least one charity or company no. I expect this should always add up to < 100%

Image

### PR DESCRIPTION
**Pull request by michaelwood**
When calculating the overview stats for /api/dashboard/overview?mode=grants the logic of fail/pass was being used to count the total number of grants which passed by just using the total number of grants that are in the source file.

This is only correct some of the time because it just so happens that if a publisher is likely to be missing a field (such as company number) then they will miss it out of all of their datasets so the calculation is coincidentally correct. Some publishers however do add such fields for _some_ of their grants which makes this calculation completely incorrect for certain metrics.

This change makes sure we use the total count from the DQT (library) rather than the aggregated data.

Updates basic test data to corrected value.

Fixes: https://github.com/ThreeSixtyGiving/datastore/issues/246

### CODE REVIEW

**Comment by michaelwood** [comment]
@mariongalley the impact of this fix is that three metrics will be corrected significantly on https://qualitydashboard.threesixtygiving.org/alldata#grants :

- "Includes recipient locations codes" goes to 65% from 80% 
- "Includes at least one charity or company no." goes to 54% from 108%
- "Includes grant duration" goes to 47% from 52%

**Comment by michaelwood** [comment]
I've created https://github.com/ThreeSixtyGiving/datastore/issues/248 as we need some worse quality test data to get better nuances on automated quality tests in the datastore for this, currently the totals generally add up to either 0% or 100% which can too easily coincidentally happen.

File: datastore/data_quality/quality_data.py [hunk 1]
```diff
@@ -324,21 +324,30 @@ def get_pc_quality_grants(self):
...
-            )
+            # Aggregate total count of errors for the metrics
+
+            # We need to drop down to SQL to do the casting/aggregates of the number of grants that are
+            # affected. Extract column/key from django orm syntax
+            # e.g. for hasRecipientIndividualsCodelists:
+            # {'quality__IndividualsCodeListsNotPresent__fail': False} extract "IndividualsCodeListsNotPresent"
+            metric_dqt_name = list(query.keys())[0].split("__")[1]
+
+            ret[metric] = self.source_file_set.annotate(
```
Comment (R2ZER0) [inline datastore/data_quality/quality_data.py]: I was confused initially what `ret[metric]` means - here is it the total number of grants that *don't* meet the quality criteria? But then below, it is changed to mean the percent of relevent grants that *do* mean the quality criteria?
If so, perhaps use a temporary var instead for the first use e.g. `metric_total_failed_grants`? then just set the actual rat at the end of the loop

Comment (michaelwood) [reply datastore/data_quality/quality_data.py]: This was how it was already however I agree should be refactored (boo hiss variables being reused for different values etc).

I've squashed this change into the commit:

```diff
diff --git a/datastore/data_quality/quality_data.py b/datastore/data_quality/quality_data.py
index 82e5229..5536f2c 100644
--- a/datastore/data_quality/quality_data.py
+++ b/datastore/data_quality/quality_data.py
@@ -332,22 +332,23 @@ class SourceFilesStats(object):
             # {'quality__IndividualsCodeListsNotPresent__fail': False} extract "IndividualsCodeListsNotPresent"
             metric_dqt_name = list(query.keys())[0].split("__")[1]
 
-            ret[metric] = self.source_file_set.annotate(
+            metric_total_errors_count = self.source_file_set.annotate(
                 total=RawSQL("((quality->%s->>'count')::numeric)", (metric_dqt_name,))
             ).aggregate(Sum("total"))["total__sum"]
 
-            if ret[metric] == None:
-                ret[metric] = 0
+            # Guard as the query can technically return None
+            if metric_total_errors_count == None:
+                metric_total_errors_count = 0
 
             # Workout percentage of total which don't have errors / all grants in the relevant recipient set
             if "Org" in metric:
                 ret[metric] = round(
-                    (total_recipient_org_grants - ret[metric])
+                    (total_recipient_org_grants - metric_total_errors_count)
                     / total_recipient_org_grants
                     * 100
                 )
             else:
-                ret[metric] = round((total_grants - ret[metric]) / total_grants * 100)
+                ret[metric] = round((total_grants - metric_total_errors_count) / total_grants * 100)
 
         return ret
```

**Review by R2ZER0 — APPROVED** [review]
Looks good :+1:

**Comment by michaelwood** [comment]
deployed

**Pull request merged**